magnum_capi_helm/common/capi_monitor.py

```diff
@@ -0,0 +1,172 @@
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+from magnum.conductor import monitors
+from magnum.objects import fields as m_fields
+
+
+class CAPIMonitor(monitors.MonitorBase):
+    def __init__(self, context, cluster, driver):
+        super(CAPIMonitor, self).__init__(context, cluster)
+        # TODO(dalees): Remove driver reference here.
+        self.driver = driver
+        self.data = {}
+        self.data["nodes"] = []
+        self.data["pods"] = []
+
+    @property
+    def metrics_spec(self):
+        # TODO(dalees): Not yet implemented for CAPI helm driver
+        return super().metrics_spec
+
+    def pull_data(self):
+        # TODO(dalees): Not yet implemented for CAPI helm driver
+        return super().pull_data()
+
+    def poll_health_status(self):
+        """Poll health status of API and nodes for given cluster from CAPI
+
+        Design Policy:
+        1.  Magnum should not require direct access to workload clusters.
+            This means all health status must come from Cluster API's
+            management cluster resources. These already provide health info.
+
+        1.  Calculating the overall health status.
+            We depend on Cluster and OpenstackCluster resources having updated
+            'status'.
+            From these, we can retrieve overall health and report back.
+
+            Some errors surface only in OpenstackCluster state, under
+            FailureMessage and FailureReason. Surface this to users, as it can
+            be helpful (eg. Neutron 409's when deleting a network that has
+            ports)
+
+            If we find any Cluster.status.conditions field *not* Status:True,
+            then list and set Unhealthy.
+
+            If we find OpenstackCluster.status.ready *not* true, report back
+            and include FailureReason and FailureMessage if set.
+
+            Iterate all nodegroup MachineDeployment and report back on
+            status.conditions.
+
+        2.  The data structure of health_status_reason.
+
+            As an attribute of the cluster, the health_status_reason have to
+            use the field type from
+            oslo.versionedobjects/blob/master/oslo_versionedobjects/fields.py
+
+        :return: None.
+
+        The class variable data is updated with current status and reason.
+
+        """
+        resource_name = self.cluster.stack_id
+        # TODO(dalees): Don't access driver internals!
+        namespace = self.driver._namespace(self.cluster)
```

Comment:
Yeah, lets move those as module methods in `magnum_capi_helm/kubernetes.py`

Reply:
Moved these shared static functions into `magnum_capi_helm/driver_utils.py`.

This cleaned up the monitor from having to use a driver reference, so this is removed.